gnd/src/main/java/com/google/android/gnd/ui/common/ResponseValidator.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.ui.common;
+
+import android.content.res.Resources;
+import com.google.android.gnd.GndApplication;
+import com.google.android.gnd.R;
+import com.google.android.gnd.model.form.Field;
+import com.google.android.gnd.model.observation.Response;
+import java8.util.Optional;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class ResponseValidator {
```

Comment:
View business logic generally belongs in ViewModels. Any reason not to include this in the AbstractFieldViewModel class?

Reply:
There were a couple of reasons to separate it out. 

 1. Right now, we are only considering if response is present or not. In future, we'll have to consider a lot of other validations as well (minLen, instanceType, regex, etc.)

 2. In the beginning, I moved all of this logic within AFVM, but it was also present in EOVM. So, to prevent duplication, I kept the original code and left it as is in favor of future cleanups. Ideally, the error should be generated and set by AFVM and EOVM should subscribe to the changes.

Reply:
> There were a couple of reasons to separate it out.
> 
> 1. Right now, we are only considering if a response is present or not. In future, we'll have to consider a lot of other validations as well (minLen, instanceType, regex, etc.). Having a small class would encourage extensive testing with all possible scenarios.

In that case, won't we end up with parallel Validator and ViewModel hierarchies for each field type?

> 2. In the beginning, I moved all of this logic within AFVM, but it was also present in EOVM. So, to prevent duplication, I kept the original code and left it as is in favor of future cleanups. Ideally, the error should be generated and set by AFVM and EOVM should subscribe to the changes.

I see.. so it sounds like it belongs in the individual AFVM?

Reply:
> In that case, won't we end up with parallel Validator and ViewModel hierarchies for each field type?

I see. So each ViewModel should contain the logic instead of aggregating them in a external class and later mapping them. Sgtm.

> I see.. so it sounds like it belongs in the individual AFVM?

Yes.

Reply:
Done b42aa16